File: skyspaceeightyeight_ui/src/services/FlightScheduleService.js
import http from "../http-common"; 

class FlightScheduleService {
  getAllFlightSchedules(searchDTO) {
    console.log(searchDTO)
    return this.getRequest(`/flightSchedule/flightSchedules`, searchDTO);
  }

  get(flightScheduleId) {
    return this.getRequest(`/flightSchedule/${flightScheduleId}`, null);
  }

  findByField(matchData) {
    return this.getRequest(`/flightSchedule?field=${matchData}`, null);
  }

  addFlightSchedule(data) {
    return http.post("/flightSchedule/addFlightSchedule", data);
  }

  update(data) {
  	return http.post("/flightSchedule/updateFlightSchedule", data);
  }
  
  uploadImage(data,flightScheduleId) {
  	return http.postForm("/flightSchedule/uploadImage/"+flightScheduleId, data);
  }




	postRequest(url, data) {
		return http.post(url, data);
      };

	getRequest(url, params) {
        return http.get(url, {
        	params: params
        });
    };

}

export default new FlightScheduleService();

File: src/main/java/com/skyspaceeightyeight/service/FlightScheduleService.java
package com.skyspaceeightyeight.service;

import java.util.List;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.data.jpa.domain.Specification;
import org.springframework.http.ResponseEntity;

import com.skyspaceeightyeight.domain.FlightSchedule;
import com.skyspaceeightyeight.dto.FlightScheduleDTO;
import com.skyspaceeightyeight.dto.FlightScheduleSearchDTO;
import com.skyspaceeightyeight.dto.FlightSchedulePageDTO;
import com.skyspaceeightyeight.dto.FlightScheduleConvertCriteriaDTO;
import com.skyspaceeightyeight.service.GenericService;
import com.skyspaceeightyeight.dto.common.RequestDTO;
import com.skyspaceeightyeight.dto.common.ResultDTO;
import java.util.List;
import java.util.Optional;





public interface FlightScheduleService extends GenericService<FlightSchedule, Integer> {

	List<FlightSchedule> findAll();

	ResultDTO addFlightSchedule(FlightScheduleDTO flightScheduleDTO, RequestDTO requestDTO);

	ResultDTO updateFlightSchedule(FlightScheduleDTO flightScheduleDTO, RequestDTO requestDTO);

    Page<FlightSchedule> getAllFlightSchedules(Pageable pageable);

    Page<FlightSchedule> getAllFlightSchedules(Specification<FlightSchedule> spec, Pageable pageable);

	ResponseEntity<FlightSchedulePageDTO> getFlightSchedules(FlightScheduleSearchDTO flightScheduleSearchDTO);
	
	List<FlightScheduleDTO> convertFlightSchedulesToFlightScheduleDTOs(List<FlightSchedule> flightSchedules, FlightScheduleConvertCriteriaDTO convertCriteria);

	FlightScheduleDTO getFlightScheduleDTOById(Integer flightScheduleId);







}





